src/app/core/services/label.service.ts

```diff
@@ -110,16 +112,27 @@ export class LabelService {
   }
 
   /**
-   * Converts the (color) hex value into RGB format
-   * @param hex: the hex value
+   * Choose black or white text color depending on background color
+   * @param bgColor: the color code of the background
    */
-  hexToRgb(hex: string) {
-    const rgbResult = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
-    return rgbResult ? {
-      r: parseInt(rgbResult[1], 16),
-      g: parseInt(rgbResult[2], 16),
-      b: parseInt(rgbResult[3], 16)
-    } : null;
+  pickTextColorBasedOnBgColor(bgColor: string) {
+    const lightColor = 'FFFFFF'; // Light text color (white)
+    const darkColor = '000000'; // Dark text color (black)
+    const color = (bgColor.charAt(0) === '#') ? bgColor.substring(1, 7) : bgColor;
+    const r = parseInt(color.substring(0, 2), 16);
+    const g = parseInt(color.substring(2, 4), 16);
+    const b = parseInt(color.substring(4, 6), 16);
+    const uiColors = [r / 255, g / 255, b / 255];
```

Comment:
In python, the convention is `r/255` (rather than `r / 255`). Is it different in JS/TS?

Reply:
I am following the tslint.json, which states that there must be whitespace for operators

Reply:
Got it.